src/main/java/org/isf/telemetry/service/remote/CustomCommonDecoder.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Open Hospital (www.open-hospital.org)
+ * Copyright © 2006-2022 Informatici Senza Frontiere (info@informaticisenzafrontiere.org)
+ *
+ * Open Hospital is a free and open source software for healthcare data management.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * https://www.gnu.org/licenses/gpl-3.0-standalone.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.isf.telemetry.service.remote;
+
+import java.io.IOException;
+
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.boot.autoconfigure.http.HttpMessageConverters;
+import org.springframework.cloud.openfeign.support.ResponseEntityDecoder;
+import org.springframework.cloud.openfeign.support.SpringDecoder;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import feign.FeignException;
+import feign.Response;
+import feign.codec.DecodeException;
+import feign.codec.Decoder;
+
+public class CustomCommonDecoder implements Decoder {
+
+	public CustomCommonDecoder() {
+		super();
+	}
+
+	@Override
+	public Object decode(Response response, java.lang.reflect.Type type) throws IOException, DecodeException, FeignException {
+		feign.Response responseFeing = (feign.Response) response;
+		HttpMessageConverter jacksonConverter = new MappingJackson2HttpMessageConverter(new ObjectMapper());
+		ObjectFactory<HttpMessageConverters> objectFactory = () -> new HttpMessageConverters(jacksonConverter);
+		return new ResponseEntityDecoder(new SpringDecoder(objectFactory)).decode(responseFeing, type);
```

Comment:
`SpringDecoder(ObjectFactory)` is deprecated in favor of `SpringDecoder(ObjectFactory, ObjectProvider)`

Reply:
it seems the SpringDecoder constructor was wrongly annotated and @deprecated wil be removed in the next versions. Indeed, if I go to the constructor definition, it has only one param:

```
	/**
	 * @deprecated in favour of
	 * {@link SpringDecoder#SpringDecoder(ObjectFactory, ObjectProvider)}
	 */
	@Deprecated
	public SpringDecoder(ObjectFactory<HttpMessageConverters> messageConverters) {
		this(messageConverters, new EmptyObjectProvider<>());
	}
```

See [here](https://github.com/spring-cloud/spring-cloud-openfeign/issues/852#issuecomment-1836465587) and [here](https://github.com/spring-cloud/spring-cloud-openfeign/commit/19d0b8188ad3b43af20156aeed86a8d526a66919).